tests/test_java.py

```diff
@@ -228,7 +225,23 @@ def sap_and_crash(self, *args, **kwargs):
     assert "memory_usage_in_bytes:" in caplog.text
 
 
-def test_already_loaded_ap_profiling_failure(tmp_path, caplog, application_pid) -> None:
+def test_already_loaded_ap_profiling_failure(tmp_path, monkeypatch, caplog, application_pid) -> None:
+    with monkeypatch.context() as m:
+        m.setattr("gprofiler.profilers.java.get_temporary_storage_path", lambda: "/tmp/fake_gprofiler_tmp")
+        with JavaProfiler(
+            1,
```

Comment:
Can keep the args in some dict and expand it twice so you don't copy them all over

Reply:
Tried but then type checking failed